python/python/tests/test_async_client.py

```diff
@@ -865,6 +865,21 @@ async def test_hrandfield_withvalues(self, redis_client: TRedisClient):
         with pytest.raises(RequestError):
             await redis_client.hrandfield_withvalues(key2, 5)
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_hstrlen(self, redis_client: TRedisClient):
+        key = get_random_string(10)
+
+        assert await redis_client.hstrlen(key, "field") == 0
+        assert await redis_client.hset(key, {"field": "value"}) == 1
+        assert await redis_client.hstrlen(key, "field") == 5
+
+        assert await redis_client.hstrlen(key, "field2") == 0
```

Comment:
not sure why you have this test, since you're already testing the non-existing field case on line 873

Reply:
line 873 is for non-existing key, I think it's nice to have both